conseil-lorre/src/test/scala/tech/cryptonomic/conseil/indexer/bitcoin/BitcoinOperationsTest.scala

```diff
@@ -49,26 +110,51 @@ class BitcoinOperationsTest
     */
   trait BitcoinOperationsStubs {
 
+    val batchConfig = BitcoinBatchFetchConfiguration(
+      indexerThreadsCount = 1,
+      httpFetchThreadsCount = 1,
+      hashBatchSize = 1,
+      blocksBatchSize = 1,
+      transactionsBatchSize = 1
+    )
+
     class BitcoinClientMock(rpcClient: RpcClient[IO]) extends BitcoinClient[IO](rpcClient)
     class BitcoinPersistenceMock extends BitcoinPersistence[IO]
 
     val bitcoinClientMock = mock[BitcoinClientMock]
     val bitcoinPersistenceMock = mock[BitcoinPersistenceMock]
     val txMock = mock[Transactor[IO]]
 
+    class DummyBitcoinOperationsMock
+        extends BitcoinOperations[IO](bitcoinClientMock, bitcoinPersistenceMock, txMock, batchConfig)
+
+    val dummyBitcoinOperationsMock = mock[DummyBitcoinOperationsMock]
+
+    /**
+      * BitcoinOperations with mocked: Bitcoin client, Persistance and transactor.
+      */
     def bitcoinOperationsStub: BitcoinOperations[IO] =
       new BitcoinOperations(
         bitcoinClient = bitcoinClientMock,
         persistence = bitcoinPersistenceMock,
         tx = txMock,
-        batchConf = BitcoinBatchFetchConfiguration(
-          indexerThreadsCount = 1,
-          httpFetchThreadsCount = 1,
-          hashBatchSize = 1,
-          blocksBatchSize = 1,
-          transactionsBatchSize = 1
-        )
+        batchConf = batchConfig
       )
+
+    /**
+      * BitcoinOperations with mocked `loadBlocksWithTransactions`.
+      * The method is used when testing the indexer startup with Depth.
+      */
+    def dummyBitcoinOperationsStub: BitcoinOperations[IO] =
```

Comment:
If I may say, here I see a lot of dummy objects very custom tailored and whose purpose I cannot easily make out, at least not from the definition site.

My approach would be to try and customize the details in the body of the test, so that it's easier to compare the testing fixtures and their specific behaviour against the specific test case logic.
You can leave the dummy definition abstract, and override the missing methods in the test body, where it's apparent what you expect from the fixture.

I have a hard time understanding what's the actual use and difference of mocking a dummy subclass, and mocking the original class... 
I'm referring to the fact that I can see a 
- DummyBitcoinOperationsMock
- dummyBitcoinOperationsMock = mock[DummyBitcoin...]

How do we make use of mocking a dummy?

Reply:
please check it now